### ISSUE TITLE
[github.io] Standardize `Core` sections into `Other` category

### ISSUE THREAD

**Issue by scbedd** [Azure/azure-sdk#4438]
This is prompted by this by PR #4411

It looks like we need to do a standardization pass on all of the main languages. End result, our `Core` github.io page should entirely disappear and be supplanted by `Other` categories that look the ones from docs.ms.com.

- [x] https://azure.github.io/azure-sdk-for-python/core.html
- [ ] https://azure.github.io/azure-sdk-for-net/core.html
- [x] https://azure.github.io/azure-sdk-for-js/core.html
- [x] https://azure.github.io/azure-sdk-for-java/core.html

Basically, we don't want to have this situation:

![image](https://user-images.githubusercontent.com/2158838/172937060-c670f6b8-5b7c-48ef-8c7e-3d3e90125cfd.png)

In _any_ of our primary languages. Everything should exist under `Other` categories like:

![image](https://user-images.githubusercontent.com/45376673/173465990-27d5288e-a68a-44ea-b993-d42c9feea2d1.png)

Properly categorizing everything starting from the `github.io` side will be the easiest way to account for this.
